hack/lib/version.sh

```diff
@@ -50,6 +50,13 @@ kube::version::get_version_vars() {
 
     # Use git describe to find the version based on annotated tags.
     if [[ -n ${KUBE_GIT_VERSION-} ]] || KUBE_GIT_VERSION=$("${git[@]}" describe --tags --abbrev=14 "${KUBE_GIT_COMMIT}^{commit}" 2>/dev/null); then
+      # This translates the "git describe" to an actual semver.org
+      # compatible semantic version that looks something like this:
+      #   v1.1.0-alpha.0.6+84c76d1142ea4d
+      #
+      # TODO: We continue calling this "git version" because so many
+      # downstream consumers are expecting it there.
+      KUBE_GIT_VERSION=$(echo "${KUBE_GIT_VERSION}" | sed "s/-\([0-9]\{1,\}\)-g\([0-9a-f]\{14\}\)$/.\1\+\2/")
```

Comment:
Yeah, that's the issue. I got this intentionally right for master because we have some needs for continuous integration, so I really wanted `v1.2.0-alpha.0-1+abcdef` < `v1.2.0`.

It's not sure how much CI is going to happen on the release branch. Arguably more should. If you look at, say, the FreeBSD model, they have a hierarchy of CURRENT / STABLE / RELEASE, where STABLE is effectively this release branch. In that world, it would be useful if the build numbers that came off the release branch looked "correct", whatever that is.

Something like `v1.0.2-2+abcdef` is compliant, but it's strictly less than `v1.0.2`, hence my comment about a second tag (we could tag `v1.0.3-prerelease`). Which, thinking out loud, I have no idea where you even put the second tag, because I killed the "dev" version commit - I could bring that back just for the purpose of tagging, but it seemed like such a wasted commit when the scripts can do it for you.